src/fundus/parser/utility.py

```diff
@@ -371,3 +392,218 @@ def parse_title_from_root(root: lxml.html.HtmlElement) -> Optional[str]:
+
+    Returns:
+        List of Images
+    """
+
+    def nodes_to_text(nodes: List[Union[lxml.html.HtmlElement, str]]) -> Optional[str]:
+        return " ".join(generic_nodes_to_text(nodes, normalize=True)) or None
+
+    for position, node, is_cover in image_nodes:
+        # parse URLs
+        urls = {}
+        if (parent := node.getparent()) is not None and parent.tag == "picture":
+            for source in parent.xpath("./source"):
+                urls.update(parse_urls_from_image(source) or {})
+        else:
+            urls.update(parse_urls_from_image(node) or {})
+        if not urls:
+            continue
+
+        # parse caption
+        caption = nodes_to_text(caption_selector(node))
+
+        # parse authors
+        authors = []
+        if isinstance(author_selector, Pattern):
+            # author is part of the caption
+            if caption and (match := re.search(author_selector, caption)):
+                authors = [match.group("credits")]
+                caption = re.sub(author_selector, "", caption).strip() or None
+        else:
+            # author is selectable as node
+            if author_nodes := author_selector(node):
+                authors = generic_nodes_to_text(author_nodes, normalize=True)
+        authors = image_author_parsing(authors, author_filter)
+
+        # parse description
+        description = nodes_to_text(alt_selector(node))
+
+        yield Image(
+            urls=urls,
+            caption=caption,
+            authors=authors,
+            description=description,
+            is_cover=is_cover,
+            position=position,
+        )
+
+
+def determine_bounds(
+    dom: DOM, paragraph_selector: XPath, upper_boundary_selector: XPath, lower_boundary_selector: Optional[XPath]
+) -> Optional[Bounds]:
+    def get_sorted_indices(nodes: List[lxml.html.HtmlElement]) -> List[int]:
+        return sorted([dom.get_index(node) for node in nodes])
+
+    # the getitem on upper_boundary_selector ensures that this throws an exception, if there are no
+    # upper_boundary_node present, as well as removing excess ones.
+    upper_boundary_nodes = [upper_boundary_selector(dom.root)[0]]
+    paragraph_nodes = paragraph_selector(dom.root)
+    lower_boundary_nodes = lower_boundary_selector(dom.root) if lower_boundary_selector else []
+
+    sorted_indices = get_sorted_indices(upper_boundary_nodes + paragraph_nodes + lower_boundary_nodes)
+
+    if len(sorted_indices) < 2:
+        return None
+
+    return Bounds(
+        upper=sorted_indices[0],
+        first_paragraph=paragraph_indices[0] if (paragraph_indices := get_sorted_indices(paragraph_nodes)) else None,
+        lower=sorted_indices[-1],
+    )
+
+
+def image_extraction(
+    doc: lxml.html.HtmlElement,
+    paragraph_selector: XPath,
+    image_selector: XPath = XPath("//figure//img"),
+    upper_boundary_selector: XPath = XPath("//main"),
+    lower_boundary_selector: Optional[XPath] = None,
```

Comment:
```suggestion
    lower_boundary_selector: Optional[XPath] = XPath("//footer"),
```
@MaxDall I would suggest adding this, because I am encountering Bounds Extraction errors regularly for premium articles. Since they often don't have any paragraphs and if there is no lower_boundary_selector set, also no lower_boundary causing the Bounds Extraction to return 0

Reply:
The only problem is that this would overwrite the last paragraph as a lower boundary. It could be all the same, could be really bad.

Having working default values from the master branch would mitigate this problem, we could also argue to not raise an exception here but instead return an empty list.

I think having `None` should be fine while working on this.